include/seqan3/alphabet/composition/union_composition.hpp

```diff
@@ -461,39 +330,202 @@ class union_composition
+        std::max({static_cast<size_t>(0), static_cast<size_t>(alphabet_size_v<alphabets_t>)...});
+
+    /*!\brief Returns an array which contains the prefix sum over all alphabet_types::value_size's.
+     * \tparam ...alphabet_types The types must satisfy seqan3::alphabet_concept.
+     *
+     * ```cpp
+     * constexpr std::array partial_sum = partial_sum_of_alphabet_sizes<dna4, gap, dna5>();
+     * assert(partial_sum.size() == 4);
+     * assert(partial_sum[0] == 0);
+     * assert(partial_sum[1] == 4);
+     * assert(partial_sum[2] == 5);
+     * assert(partial_sum[3] == 10);
+     * ```
+     */
+    template <typename ...alphabets_t>
+    //!\cond
+        requires (alphabet_concept<alphabets_t> && ...)
+    //!\endcond
+    static constexpr auto partial_sum_of_alphabet_sizes() noexcept
+    {
+        constexpr size_t value_size = sum_of_alphabet_sizes_v<alphabets_t...>;
+        using rank_t = detail::min_viable_uint_t<value_size>;
+
+        constexpr size_t N = sizeof...(alphabets_t) + 1;
+        using array_t = std::array<rank_t, N>;
+
+        array_t partial_sum{0, alphabet_size_v<alphabets_t>...};
+        for (size_t i = 1u; i < N; ++i)
+            partial_sum[i] += partial_sum[i-1];
+
+        return partial_sum;
+    }
+
+    /*!\brief Returns an fixed sized map at compile time where the key is the rank
+     * of alphabet_t and the value is the corresponding char of that rank.
+     * \sa value_to_char_table
+     *
+     * ```cpp
+     * constexpr std::array table1 = make_value_to_char_table<5, char>(dna4{});
+     * assert(table1.size() == 5);
+     * assert(table1[0] == 'A');
+     * assert(table1[1] == 'C');
+     * assert(table1[2] == 'G');
+     * assert(table1[3] == 'T');
+     * assert(table1[4] == '\0');
+     * ```
+     */
+    template <size_t table_size, typename alphabet_t>
+        requires (alphabet_size_v<alphabet_t> <= table_size)
+    static constexpr auto make_value_to_char_table(alphabet_t alphabet) noexcept
+    {
+        using array_t = std::array<char_type, table_size>;
+        array_t value_to_char{};
+
+        for (size_t i = 0u; i < alphabet_size_v<alphabet_t>; ++i)
+            value_to_char[i] = alphabet.assign_rank(i).to_char();
+
+        return value_to_char;
+    }
+
+    /*!\brief Returns a map at compile time where the key is the rank of the union
+     * of all alphabets and the value is the corresponding char of that rank and
+     * alphabet.
+     *
+     * ```cpp
+     * constexpr std::array value_to_char = value_to_char_table<char, dna4, gap, dna5>();
+     * assert(value_to_char.size() == 10);
+     * assert(value_to_char[0] == 'A');
+     * assert(value_to_char[1] == 'C');
+     * assert(value_to_char[2] == 'G');
+     * assert(value_to_char[3] == 'T');
+     * assert(value_to_char[4] == '-');
+     * assert(value_to_char[5] == 'A');
+     * assert(value_to_char[6] == 'C');
+     * // and so on
+     * ```
+     */
+    static constexpr auto value_to_char_table() noexcept
```

Comment:
This is called `value_to_char_table`, but is actually a function, not a table :thinking: 
Also the previous function actually doesn't "make" this, it is the other way around, the previous function is used by this one.

Reply:
Also the documentation of this actually covers `value_to_char`...

Reply:
:tada: Professional attitude to :man_facepalming: :children_crossing:

This one is there to make the `value_to_char_table` of the `union_composition` and uses `make_value_to_char_table` in a expansion expression for every alphabet.